runtime/calamari/src/lib.rs

```diff
@@ -750,8 +775,48 @@ pub type Executive = frame_executive::Executive<
     frame_system::ChainContext<Runtime>,
     Runtime,
     AllPalletsReversedWithSystemFirst,
-    (),
+    UpgradeSessionKeys,
 >;
+// When this is removed, should also remove `OldSessionKeys`.
+pub struct UpgradeSessionKeys;
+impl frame_support::traits::OnRuntimeUpgrade for UpgradeSessionKeys {
+    fn on_runtime_upgrade() -> frame_support::weights::Weight {
+        use opaque::transform_session_keys;
+        // transform_session_keys runs translate() on NextKeys and on QueuedKeys which is (at worst or faster than) 1 read 1 write
+        let validator_set_len: u64 = Session::queued_keys().len().try_into().unwrap();
```

Comment:
I suppose it will never panic our runtime, right?

Reply:
the session keys DB on KMA is <30 entries
and the upgrade reserves half a block for good measure ( if it takes longer, it uses the actual weight )
30 db reads/writes should not have a chance to overflow block execution limits